src/lang/errors.ts

```diff
@@ -151,3 +154,33 @@ export function kclErrorsToDiagnostics(
     return sourceRanges
   })
 }
+
+export function complilationErrorsToDiagnostics(
+  errors: CompilationError[]
+): CodeMirrorDiagnostic[] {
+  return errors?.map((err) => {
+    let severity: any = 'error'
+    if (err.severity === 'Warning') {
+      severity = 'warning'
+    }
+    let actions
+    const suggestion = err.suggestion
+    if (suggestion) {
+      actions = [
+        {
+          name: suggestion.title,
+          apply: (view: EditorView, from: number, to: number) => {
+            view.dispatch({ changes: { from, to, insert: suggestion.insert } })
+          },
+        },
+      ]
+    }
+    return {
+      from: err.sourceRange[0],
+      to: err.sourceRange[1],
+      message: err.message,
```

Comment:
Did you test this with a diagnostic in an imported file? I'm afraid we won't be able to handle it yet. If that's the case, filtering for only moduleId == 0 might be an option.

Reply:
Currently there is no module id in the TS encoding of a SourceRange and I don't know how the old code is getting it (because it is using the serialisation directly, I think). Unfortunately SourceRange is used all over the place, so changing it is a pain

Reply:
Module ID is the third thing in the source range tuple. There are some places in TS where we didn't update the type to indicate that it has three members, but in this TS module, we do.

My fault. I was skimming this PR the other night and I didn't realize that `err.sourceRange[2]` _is_ checked at the top of this function.